librz/main/rz-diff.c

```diff
@@ -1,1305 +1,1454 @@
+
+	rz_config_set_i(core->config, "scr.color", 0);
+	rz_config_set_b(core->config, "scr.interactive", false);
+	rz_config_set_b(core->config, "cfg.debug", false);
+	core->print->scr_prompt = false;
+
+#if __WINDOWS__
+	char *winpath = rz_acp_to_utf8(filename);
+	cfile = rz_core_file_open(core, winpath, 0, 0);
+	free(winpath);
+#else
+	cfile = rz_core_file_open(core, filename, 0, 0);
+#endif
+	if (!cfile) {
+		rz_diff_error("cannot open file '%s'\n", filename);
+		goto rz_diff_load_file_with_core_fail;
 	}
-	rz_cons_break_push(NULL, NULL);
-	for (i = 0; i < sz; i += w) {
-		if (rz_cons_is_breaked()) {
-			break;
-		}
-		if (i + w >= sz) {
-			pad = w - sz + i;
-			w = sz - i;
-		}
-		bool eq = !memcmp(a + i, b + i, w);
-		if (eq) {
-			ctx--;
-			if (ctx == -1) {
-				rz_cons_printf("...\n");
-				continue;
-			}
-			if (ctx < 0) {
-				ctx = -1;
-				continue;
-			}
-		} else {
-			ctx = DUMP_CONTEXT;
-		}
-		rz_cons_printf(eq ? Color_GREEN : Color_RED);
-		rz_cons_printf("0x%08x%c ", i, eq ? ' ' : '!');
-		rz_cons_printf(Color_RESET);
-		for (j = 0; j < w; j++) {
-			bool eq2 = a[i + j] == b[i + j];
-			if (!eq) {
-				rz_cons_printf(eq2 ? Color_GREEN : Color_RED);
-			}
-			rz_cons_printf("%02x", a[i + j]);
-			if (!eq) {
-				rz_cons_printf(Color_RESET);
-			}
-		}
-		for (j = 0; j < pad; j++) {
-			rz_cons_printf("  ");
-		}
-		rz_cons_printf(" ");
-		for (j = 0; j < w; j++) {
-			bool eq2 = a[i + j] == b[i + j];
-			if (!eq) {
-				rz_cons_printf(eq2 ? Color_GREEN : Color_RED);
-			}
-			rz_cons_printf("%c", IS_PRINTABLE(a[i + j]) ? a[i + j] : '.');
-			if (!eq) {
-				rz_cons_printf(Color_RESET);
-			}
-		}
-		for (j = 0; j < pad; j++) {
-			rz_cons_printf(" ");
-		}
-		rz_cons_printf("   ");
-		for (j = 0; j < w; j++) {
-			bool eq2 = a[i + j] == b[i + j];
-			if (!eq) {
-				rz_cons_printf(eq2 ? Color_GREEN : Color_RED);
-			}
-			rz_cons_printf("%02x", b[i + j]);
-			if (!eq) {
-				rz_cons_printf(Color_RESET);
-			}
-		}
-		for (j = 0; j < pad; j++) {
-			rz_cons_printf("  ");
-		}
-		rz_cons_printf(" ");
-		for (j = 0; j < w; j++) {
-			bool eq2 = a[i + j] == b[i + j];
-			if (!eq) {
-				rz_cons_printf(eq2 ? Color_GREEN : Color_RED);
-			}
-			rz_cons_printf("%c", IS_PRINTABLE(b[i + j]) ? b[i + j] : '.');
-			if (!eq) {
-				rz_cons_printf(Color_RESET);
-			}
-		}
-		rz_cons_printf("\n");
-		rz_cons_flush();
+
+	if (!rz_core_bin_load(core, NULL, UT64_MAX)) {
+		rz_diff_error("cannot load file '%s'\n", filename);
+		goto rz_diff_load_file_with_core_fail;
 	}
-	rz_cons_break_end();
-	rz_cons_printf("\n" Color_RESET);
-	rz_cons_flush();
-	if (as != bs) {
-		rz_cons_printf("...\n");
+
+	if (!rz_core_bin_update_arch_bits(core)) {
+		rz_diff_error("cannot set architecture with bits\n");
+		goto rz_diff_load_file_with_core_fail;
 	}
+
+	bfile = core_get_file(cfile);
+	if (!bfile) {
+		rz_diff_error("cannot get architecture with bits\n");
+		goto rz_diff_load_file_with_core_fail;
+	}
+
+	RzList *sections = rz_bin_maps_of_file_sections(bfile);
```

Comment:
`rz_bin_maps_of_file_sections()` is not correct here, it is a special function that bin plugins can use. Here, you should be able to use just the already populated `maps` member of `RzBinObject`.